chall05/vde-dios.c

```diff
@@ -0,0 +1,31 @@
+char   ft_lower(char letter)
+{
+    if (letter >= 'A' && letter <= 'Z')
+        return (letter + 32);
+    return (letter);
+}
+
+int ft_ie_except_after_c(char* str)
+{
+    char *p;
+
+    if (str)
```

Comment:
It is my opinion, but you don't have to protect from `NULL` when making such a function. A function is not like a program, a function CAN crash. Let me explain myself:  
Here this function has a defined behavior:
- It takes a word as input (lowercased although I think we forgot to explicitly specify it)
- It MUST return `1` when the rule is respected
- It MUST return `0` when the rule is not respected

But it is not specified what you must or should do when the parameter isn't allocated, or isn't a word... So then, if an unallocated string causes a segmentation fault, it is not your problem, the user should have passed an allocated string.  
Which brings me to this `NULL` correspond to `0x0`, you neither can write or read the data present there. So why would you protect this specific memory zone? And not another like `0x1` which can't be written nor read either. But this is yet a subjective point of view.  
To be more objective, `NULL` is not a word, it isn't even a string! So how could you decide if it should return `0` or `1`? Both return values would be correct, as it is an undefined behavior. And well, crashing is also correct, in an undefined behavior.

Beware, what I am saying only applies to separated functions like this, when you do a program, a website, a videogame, or anything of the sort, there should not be undefined behaviors, because then it would be a bug!